src/services/counters/CounterService.ts

```diff
@@ -52,42 +52,54 @@ export class CounterService {
   }
 
   getAllCountersHelper(cacheContainer: ICacheContainer): ng.IPromise<ICounter[]> {
-
+    let request: IDmrRequest;
     let deferred: ng.IDeferred<ICounter[]> = this.$q.defer<ICounter[]>();
-    this.jGroupsService.getServerGroupCoordinator(cacheContainer.serverGroup).then((coord: IServerAddress) => {
-
-      let request: IDmrRequest = <IDmrRequest>{
-        address: this.getCountersRuntimeAddress(cacheContainer.name, coord.host, coord.name),
+    if (this.launchType.isStandaloneMode()) {
+      request = <IDmrRequest>{
+        address: this.getCountersRuntimeAddress(cacheContainer.name),
         "recursive-depth": 2,
         "include-runtime": true
       };
-
-      this.dmrService.readResource(request)
-        .then((response: any): ICounter[] => {
-          let counters: ICounter[] = [];
-          let trail: String [] = [];
-          traverse(response, (key: string, value: string, trail: string []) => {
-            let traversingCounterObject: boolean = (key === "name");
-            if (traversingCounterObject) {
-              let counter: any = deepValue(response, trail);
-              let type: string = trail[0];
-              if (isNotNullOrUndefined(counter) && type === "strong-counter") {
-                counters.push(new StrongCounter(counter.name, counter.storage, counter["initial-value"],
-                  counter.value, counter["lower-bound"], counter["upper-bound"]));
-              } else if (isNotNullOrUndefined(counter) && type === "weak-counter") {
-                counters.push(new WeakCounter(counter.name, counter.storage, counter["initial-value"],
-                  counter.value, counter.concurrency));
-              }
-            }
-          }, trail);
-          return counters;
-        }).then(counters => {
+      this.findCounters(request).then(counters => {
         deferred.resolve(counters);
       });
-    });
+    } else {
+      this.jGroupsService.getServerGroupCoordinator(cacheContainer.serverGroup).then((coord: IServerAddress) => {
```

Comment:
No need for typing here, can just be `coord => {`

Reply:
Ok, I see what you mean. It is easier for me to think about the object, functions, and code in general if it's type is declared.